src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsController.java

```diff
@@ -185,12 +190,26 @@ public TaskActionsController(CamundaService camundaService,
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @PostMapping(path = "/{task-id}/complete")
     public ResponseEntity<Void> completeTask(@RequestHeader("Authorization") String authToken,
-                                             @PathVariable(TASK_ID) String taskId) {
+                                             @RequestHeader("ServiceAuthorization") String serviceAuthToken,
+                                             @PathVariable(TASK_ID) String taskId,
+                                             @RequestBody(required = false) CompleteTaskRequest completeTaskRequest) {
         List<PermissionTypes> endpointPermissionsRequired = asList(OWN, EXECUTE);
 
         AccessControlResponse accessControlResponse = accessControlService.getRoles(authToken);
 
-        camundaService.completeTask(taskId, accessControlResponse, endpointPermissionsRequired);
+        boolean isPrivilegedRequest =
+            privilegedAccessControlService.hasPrivilegedAccess(serviceAuthToken, accessControlResponse);
+
+        if (isPrivilegedRequest && completeTaskRequest != null && completeTaskRequest.getCompletionOptions() != null) {
```

Comment:
Can we add the unit test to cover this logic? Sonar has flagged it. Whilst it may be covered in ft, Its also essential to have unit test coverage as we maintain the bottom up approach for testing standard